### PR TITLE
Fix history for #16

### PR DESCRIPTION
**Pull request by caasi**
- remove unused `setState`,
- merge keys and deal with empty states

### CODE REVIEW

File: src/stores/EntryStore.ts [hunk 1]
```diff
@@ -101,7 +109,8 @@ export default class EntryStore extends EventEmitter {
   public serialize():string {
     var params = [];
     for (let key in this.tracks) {
-      params[key] = this.tracks[key].join(',');
+      let param = this.tracks[key].join(',');
+      params[key] = param || undefined;
```
Comment (c9s) [inline src/stores/EntryStore.ts:115]: is this necessary ?

File: src/stores/EntryStore.ts [hunk 2]
```diff
@@ -89,8 +89,16 @@ export default class EntryStore extends EventEmitter {
...
-      this.tracks[key] = params[key].split(/,/);
+    let pkeys = Object.keys(params);
+    let keys = Object.keys(this.tracks);
+    for (let k of pkeys) {
+      if (keys.indexOf(k) === -1) {
+        keys.push(k);
+      }
+    }
+    for (let key of keys) {
+      let track = params[key] || '';
```
Comment (c9s) [inline src/stores/EntryStore.ts]: `params[key]` can only be string or undefined, so it's not necessary to give it an empty string.

Comment (c9s) [inline src/stores/EntryStore.ts]: `typeof params[key] === "string"` will work.

File: src/stores/EntryStore.ts [hunk 3]
```diff
@@ -89,8 +89,16 @@ export default class EntryStore extends EventEmitter {
 
   public loadQueryString(qs:string) {
     let params = querystring.parse(location.search);
-    for (let key in params) {
-      this.tracks[key] = params[key].split(/,/);
+    let pkeys = Object.keys(params);
```
Comment (c9s) [inline src/stores/EntryStore.ts]: To avoid confusion, renaming `pkeys` to `paramNames` is better.

File: src/stores/EntryStore.ts (hunk 1 above)
Comment (caasi) [inline src/stores/EntryStore.ts:115]: click "是" in Q2 and click "回上一題" then one will get `http://localhost:8080/?11=` without setting `params[key] = undefined`

**Comment by caasi** [comment]
updated~

File: src/stores/EntryStore.ts (hunk 1 above)
Comment (c9s) [inline src/stores/EntryStore.ts:115]: got !

**Comment by c9s** [comment]
awesome great job!

**Pull request merged**